### PR TITLE
Add docs and man page entry for --volume-driver

### LINKED ISSUE #18070: [doc] document --volume-driver in man page
I'm using the debian package from https://apt.dockerproject.org/repo/ on version 1.9.0-0~vivid and the --volume-driver option is not documented in the man page. It appears in the ```docker run --help``` but not in ```man docker run```.

### PR DESCRIPTION
**Pull request by bfirsh**
fixes #18070

Signed-off-by: Ben Firshman ben@firshman.co.uk

### CODE REVIEW

File: man/docker-run.1.md [hunk 1]
```diff
@@ -478,6 +479,13 @@ For example, you can specify either `/foo` or `foo` for a `host-dir` value.
 If you supply the `/foo` value, Docker creates a bind-mount. If you supply
 the `foo` specification, Docker creates a named volume.
 
+**--volume-driver**=""
+   Optional volume driver for the container
+
+   When using the `-v` flag to add mounts to a container, a driver can be
+   specified to create the volumes with. See **docker-volume-create(1)** for
+   full details.
```
Comment (thaJeztah) [inline man/docker-run.1.md]: Perhaps we should mention that this driver will also be used for any `VOLUME`

Comment (thaJeztah) [inline man/docker-run.1.md]: Also, can you add an example to `docs/reference/commandline/run.md`?

**Comment by thaJeztah** [comment]
Actually surprised this wasn't in there, probably because it was implemented as "experimental" first.

Looks like it also never made it into the completion scripts :disappointed_relieved:

/cc @sdurrheimer @albers

**Comment by sdurrheimer** [comment]
This should also be in `docs/reference/commandline/create.md` I guess.

**Comment by thaJeztah** [comment]
Yes, you're right @bfirsh ^^

**Comment by albers** [comment]
WRT completions: are the possible values for `--volume-driver` the same as for `docker volume create --driver`, i.e. just `local` (as long as there is no API for enumerating installed plugins)?

**Comment by thaJeztah** [comment]
@albers yes it should be the same

**Comment by albers** [comment]
@thaJeztah Thanks. I'll add this to completion once this PR is merged.

**Comment by LK4D4** [comment]
ping @thaJeztah @moxiegirl

File: docs/reference/commandline/create.md [hunk 2]
```diff
@@ -69,6 +69,7 @@ Creates a new container.
       --ulimit=[]                   Ulimit options
       --uts=""                      UTS namespace to use
       -v, --volume=[]               Bind mount a volume
+      --volume-driver=""            Optional volume driver for the container
```
Comment (moxiegirl) [inline docs/reference/commandline/create.md]: Is `Optional` needed here? These are all options on this list.

Container's volume driver.

File: docs/reference/commandline/run.md [hunk 3]
```diff
@@ -75,6 +75,7 @@ parent = "smn_cli"
       --ulimit=[]                   Ulimit options
       --uts=""                      UTS namespace to use
       -v, --volume=[]               Bind mount a volume
+      --volume-driver=""            Optional volume driver for the container
```
Comment (moxiegirl) [inline docs/reference/commandline/run.md]: same comment here

File: man/docker-create.1.md [hunk 4]
```diff
@@ -271,6 +272,13 @@ This value should always larger than **-m**, so you should always use this with
 **-v**, **--volume**=[]
    Bind mount a volume (e.g., from the host: -v /host:/container, from Docker: -v /container)
 
+**--volume-driver**=""
```
Comment (moxiegirl) [inline man/docker-create.1.md]: Don't think we need two lines here:

Container's volume driver. This driver creates volumes specified either from a
Dockerfile's `VOLUME` instruction or from the `docker run -v` flag.See
docker-volume-create(1) for full details.

File: man/docker-run.1.md [hunk 5]
```diff
@@ -478,6 +479,13 @@ For example, you can specify either `/foo` or `foo` for a `host-dir` value.
 If you supply the `/foo` value, Docker creates a bind-mount. If you supply
 the `foo` specification, Docker creates a named volume.
 
+**--volume-driver**=""
```
Comment (moxiegirl) [inline man/docker-run.1.md]: Container's volume driver. This driver creates volumes specified either from a
Dockerfile's `VOLUME` instruction or from the `docker run -v` flag.See
docker-volume-create(1) for full details.

**Comment by thaJeztah** [comment]
ping @bfirsh can you update the PR with the comments?

**Comment by thaJeztah** [comment]
carried in https://github.com/docker/docker/issues/18394, so I'll close this one

**Pull request closed without merging**